app/code/Magento/CatalogGraphQl/Model/Resolver/Categories/DataProvider/Category/CollectionProcessor/CatalogProcessor.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Categories\DataProvider\Category\CollectionProcessor;
+
+use Magento\Catalog\Model\ResourceModel\Category\Collection;
+use Magento\CatalogGraphQl\Model\Resolver\Categories\DataProvider\Category\CollectionProcessorInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\GraphQl\Model\Query\ContextInterface;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface as SearchCriteriaCollectionProcessor;
+
+/**
+ * Apply pre-defined catalog filtering
+ *
+ * {@inheritdoc}
+ */
+class CatalogProcessor implements CollectionProcessorInterface
+{
+    /** @var SearchCriteriaCollectionProcessor */
+    private $collectionProcessor;
+
+    public function __construct(
+        SearchCriteriaCollectionProcessor $collectionProcessor
+    ) {
+        $this->collectionProcessor = $collectionProcessor;
+    }
+
+    /**
+     * Process collection to add additional joins, attributes, and clauses to a category collection.
+     *
+     * @param Collection $collection
+     * @param SearchCriteriaInterface $searchCriteria
+     * @param ContextInterface|null $context
+     * @return Collection
+     */
+    public function process(
+        Collection $collection,
+        SearchCriteriaInterface $searchCriteria,
+        ContextInterface $context = null
+    ): Collection
+    {
```

Comment:
This { should be on previous line

Reply:
hey @BarnyShergold 

the changes to core methods are with the purpose to expose modular collection processing, so other layers in the application can have a say what the collection consists of.
Check maybe implementation of "products" query for an example - it is the same idea.

On a closer look, please notice, there is no bug to be solved, or new requirement to be satisfied, just a refactor, thus, should be sufficient that already present api-functinal tests do not break (and they don't :) ), no need for new ones.

In more details, I would bring your attention to GraphQl/Catalog tests, and there all the Category*.Test.php and all tests from CategoriesQuery folder are the top relevant to verify the refactor.

Anyway, bigger picture is here: https://github.com/magento/magento2/issues/28550, if it helps to gain depth on the idea that is running in this PR. If any questions, just contact me... I can give you even more details.

Reply:
@pmarjan - Your reasoning is sound - just wanted to check!
You just need to fix the minor format issues and the failing tests and all will be good.